### ISSUE TITLE
[feat] Add support for a configuration file

### ISSUE THREAD

**Issue by VJMReichenbach** [WGUNDERWOOD/tex-fmt#39]
I personally don't want my lines to wrapped and always adding `--keep` is a bit annoying (and I often forget it). It would be nice, if there was a configuration file for both system wide config (e.g. `~/.config/.tex-fmt.toml` and/or `~/.tex-fmt.toml`) and a project specific config (e.g. `~/my_folder/.tex-fmt.toml`) where you configure the formatter instead of using cli flags.
It would also make it possible for two people to use tex-fmt on the same project with the same settings, while using different editors, without any editor configuration.

For reference, I like the way [StyLua](https://github.com/JohnnyMorganz/StyLua?tab=readme-ov-file#finding-the-configuration) does it.

**Comment by WGUNDERWOOD** [comment]
Thanks for this suggestion! I would be interested in the possibility of adding support for a configuration file. However there are some issues/concerns to be addressed first.

- My aim for tex-fmt is that configuration should remain relatively limited, for simplicity of use, consistency of output, and maintainability.
- Configuration files increase the complexity of distributing the software, as they would need to work on multiple operating systems.
- Relying on configuration files can make the tool less portable, e.g. if you are working on another machine and don't have access to the config file.

**Comment by VJMReichenbach** [comment]
> My aim for tex-fmt is that configuration should remain relatively limited, for simplicity of use, consistency of output, and maintainability.

I really like your default configuration. I just don't want tex-fmt to wrap lines, so that is the only change I personally would make. The only thing I would use the config file for is setting options I can already change with cli flags.
I guess that would even improve consistency of the output since you don't have to share your specific command you used to format. Instead you can only share the repository's  config file.

> Configuration files increase the complexity of distributing the software, as they would need to work on multiple operating systems.

Valid point. I still think at least local config files should be supported, since you only have to check the current directory. (I've never written rust code on e.g. Windows, so if there is a fundamental difference between using a relative path on Windows and e.g. Linux I understand your concern.) 

I've never used it, but [confy](https://github.com/rust-cli/confy) is a crate that handles all the operating system stuff for you. I just ran the example and it seems fairly easy to implement a global config that way.

> Relying on configuration files can make the tool less portable, e.g. if you are working on another machine and don't have access to the config file.

If you can share your code, shouldn't you be able to share your config? I would just put an `.tex-fmt.toml` file in the root of my repo. That way, everybody who has access to the code, also has access to the config.

**Comment by VJMReichenbach** [comment]
Maybe using [directories](https://github.com/dirs-dev/directories-rs) to handle operating system specific paths in combination with [figment](https://github.com/SergioBenitez/Figment) for handling global and local configs is an option?

**Comment by WGUNDERWOOD** [comment]
Thanks for these thoughts, and for supplying some useful links! I agree with you that the config file should only contain arguments already available on the CLI. I'll take a closer look at the config file crates to decide which looks best.

**Comment by VJMReichenbach** [comment]
Will close this, once the `config` branch is merged

**Comment by WGUNDERWOOD** [comment]
Thanks, I am closing this since `config` is merged into `develop` and will merge into main soon. Let me know if you are able to test it out before -- it would be good to know if it works as expected for you.

**Comment by VJMReichenbach** [comment]
The global config file doesn't work for me, but everything else works as expected.
I have a file at `~/.config/tex-fmt/tex-fmt.toml`, but the options don't get recognized. For testing I modified `main.rs` to print the `args` variable and quit afterwards (but as far as I understand this shouldn't change anything.)
The modified section looks like this:
```rust
fn main() -> ExitCode {
    let mut args = get_args();
    init_logger(args.verbosity);

    let mut logs = Vec::<Log>::new();
    let mut exit_code = args.resolve(&mut logs);
    println!("{:?}", args);
    return ExitCode::from(1);

    if exit_code == 0 {
        if args.stdin {
 ... More code here
```
Does this maybe affect my testing?

A local config in the same directory is picked up as expected and it's contents are overwritten with Cli arguments as expected.
Btw if I specify my global config file with `--config` it works as expected.

**Comment by WGUNDERWOOD** [comment]
I have changed the implementation slightly so now `args::Args` should contain the resolved config path. Perhaps you can try this again on `develop` and post your modified `main.rs`, `tex-fmt.toml` and output from tex-fmt? I am not currently able to replicate this but would like to understand what the issue is.

**Comment by VJMReichenbach** [comment]
Ok I am stupid. I was executing tex-fmt by executing the build binary in the `target/release` directory. But of course that meant that it picked up the `tex-fmt.toml` file in the git root.
So everything worked as expected. If i delete the `tex-fmt.toml` in the git root, the options from my global config file are picked up correctly.
So it was user error.

For the future: 
- Maybe move the `tex-fmt.toml` file from the git root somewhere else? Like the `man` folder?
- Maybe add an option for printing all the arguments and exit, for debugging?
  - If you want to implement the option above, maybe print all detected config files and its options too?
